website/source/docs/providers/aws/r/default_network_acl.html.markdown

```diff
@@ -0,0 +1,151 @@
+---
+layout: "aws"
+page_title: "AWS: aws_default_network_acl"
+sidebar_current: "docs-aws-resource-default-network-acl"
+description: |-
+  Manage the default Network ACL resource.
+---
+
+# aws\_default\_network\_acl
+
+Provides a resource to manage the default AWS Network ACL. VPC Only.
+
+Each VPC created in AWS comes with a Default Network ACL that can be managed, but not
+destroyed. **This is an advanced resource**, and has special caveats to be aware
+of when using it. Please read this document in its entirety before using this
+resource.
+
+The `aws_default_network_acl` behaves differently from normal resources, in that
+Terraform does not _create_ this resource, but instead attempts to "adopt" it
+into management. We can do this because each VPC created has a Default Network
+ACL that cannot be destroyed, and is created with a known set of default rules. 
+
+When Terraform first adopts the Default Network ACL, it **immediately removes all
+rules in the ACL**. It then proceeds to create any rules specified in the 
+configuration. This step is required so that only the rules specified in the 
+configuration are created.
+
+For more information about Network ACLs, see the AWS Documentation on 
+[Network ACLs][aws-network-acls].
+
+## Basic Example Usage, with default rules
+
+```
+resource "aws_vpc" "mainvpc" {
+  cidr_block = "10.1.0.0/16"
+}
+
+resource "aws_default_network_acl" "default" {
+  default_network_acl_id = "${aws_vpc.mainvpc.default_network_acl_id}"
+
+  ingress {
+    protocol   = -1
+    rule_no    = 100
+    action     = "allow"
+    cidr_block = "0.0.0.0/0"
+    from_port  = 0
+    to_port    = 0
+  }
+
+  egress {
+    protocol   = -1
+    rule_no    = 100
+    action     = "allow"
+    cidr_block = "0.0.0.0/0"
+    from_port  = 0
+    to_port    = 0
+  }
+}
+```
+
+## Example config to deny all Egress traffic, allowing Ingress
+
+```
+resource "aws_vpc" "mainvpc" {
+  cidr_block = "10.1.0.0/16"
+}
+
+resource "aws_default_network_acl" "default" {
+  default_network_acl_id = "${aws_vpc.mainvpc.default_network_acl_id}"
+
+  ingress {
+    protocol   = -1
+    rule_no    = 100
+    action     = "allow"
+    cidr_block = "0.0.0.0/0"
+    from_port  = 0
+    to_port    = 0
+  }
+
+}
+```
```

Comment:
A totally empty config will deny all, right? Might be worth calling that out as a third example with a description something like:

> This config denies all traffic in the Default ACL. This can be useful if you want a locked down default to force all resources in the VPC to assign a non-default ACL.